build.sbt

```diff
@@ -184,21 +185,51 @@ lazy val libSettings =
     )
   )
 
-lazy val gcSettings =
-  if (!System.getenv.containsKey("SCALANATIVE_GC")) {
-    println("Using default gc")
-    Seq.empty
-  } else {
-    val gc = System.getenv.get("SCALANATIVE_GC")
-    println(s"Using gc based on SCALANATIVE_GC=$gc")
-    Seq(nativeGC := gc)
+lazy val gcSetting =
+  nativeGC := {
+    val log     = sLog.value
+    val default = (nativeGC in ThisBuild).value
+    val gc      = Option(System.getenv.get("SCALANATIVE_GC")).getOrElse(default)
+    log.info(s"Using $gc gc")
+    gc
   }
 
 lazy val projectSettings =
   ScalaNativePlugin.projectSettings ++ Seq(
     scalaVersion := libScalaVersion,
     resolvers := Nil
-  ) ++ gcSettings
+  )
+
+lazy val scalaNative =
+  project
+    .in(file("."))
+    .aggregate(
+      util,
+      nir,
+      tools,
+      nscplugin,
+      sbtScalaNative,
+      nativelib,
+      javalib,
+      auxlib,
+      scalalib,
+      testingCompilerInterface,
+      testingCompiler,
+      testInterface,
+      testInterfaceSerialization,
+      testInterfaceSbtDefs,
+      sandbox,
+      demoNative,
+      tests,
+      demoJVM,
+      benchmarks
+    )
+    .settings(
+      name := "scala-native",
+      gcSetting,
+      aggregate in publishLocal := false
+    )
+    .settings(noPublishSettings)
```

Comment:
Lets *not* use aggregates. They are confusing in a number of ways. One should think of Scala Native a collection of smaller projects, not a single mega-project.

Reply:
I agree that aggregates are a bit messy looking but a `default` main project is like "aggregate all" by default and it publishes a default project in the current build. Also, if I put the gcSetting in project settings rather than the root project the setting gets set three times and repeats the message.